### ISSUE TITLE
Different behaviour when using dbWriteTable() on Snowflake

### ISSUE THREAD

**Issue by jjimenezRe** [r-dbi/odbc#833]
We have been using odbc::dbWriteTable() to write data to Snowflake tables.

Until 1.5.0 (including 1.4.2), our data CSV files (7+ million records) were compressed and sent as different files to Snowflake, then inserted with a single INSERT statement:

![image](https://github.com/user-attachments/assets/268ab63c-f8fb-46b5-bbcf-40b3a1907776)

Starting with 1.5.0 and using default settings, it does INSERT of 1024 records each:

![image](https://github.com/user-attachments/assets/a538d08a-4be9-44e2-ba67-1f9952a69f0f)

![image](https://github.com/user-attachments/assets/f297fa32-555a-4951-b1a2-68f12e220556)

We noticed in the release changelist a parameter called "odbc.batch_rows", which we were not using previously.

When setting this parameter to big numbers (tried a few, 10.000, 100.000, and 1 million), we noticed the data was being sent to Snowflake compressed again and inserted in X rows (whatever number we used in odbc.batch_rows):

![image](https://github.com/user-attachments/assets/06980edc-3d17-4cb5-bc2e-1f7efba5611e)

The pattern being slightly different as its not a single INSERT anymore, but a PUT/INSERT combination until all the data is sent.

Is this the intended/expected behavior? Is there any way to go back to the previous behavior, or is this supposed to be a better performance?

**Comment by KeyboardSourcerer** [comment]
Note: writing a 7 million records dataset with 18 attributes from Windows vanilla R 4.1.1, with ODBC 1.5.0, results in desired behavior (packaging data to compressed files, uploading to a temporary stage, and then inserting) instead of the direct INSERTs of 1024 records at a time. Thus, the issue appears isolated to Linux variants of R.

**Comment by lalvesRe** [comment]
Note: writing a 7 million records dataset with 18 attributes from Windows vanilla R 4.1.1, with ODBC 1.5.0, results in desired behavior (packaging data to compressed files, uploading to a temporary stage, and then inserting) instead of the direct INSERTs of 1024 records at a time. The issue appears isolated to Linux variants of R.

**Comment by detule** [comment]
Hello:

Thanks for the investigation.

The technique we use to write to target tables is described on [this page](https://docs.snowflake.com/en/developer-guide/odbc/odbc-using#binding-parameters-to-array-variables-for-batch-inserts).  Noticed on that page:

```
When you use this technique to insert a large number of values, the driver can improve performance by streaming
the data (without creating files on the local machine) to a temporary stage for ingestion. The driver automatically
does this when the number of values exceeds a threshold.
```

Prior to https://github.com/r-dbi/odbc/commit/6f6d4e782e9f4e040ca38c88796af7301fb0138e we would execute essentially one ```SQLExecDirect``` after binding data to buffers for all the columns and all of the rows.  There were some concerns that this operation may have a large footprint and create memory pressure on the host.  After that commit the default is to partition the data in chunks of ~1K rows and send each of them.

To your questions:
* No - I wouldn't expect that commit to be a performance enhancement ( but hopefully it helps with memory management ).
* Users that may be more sensitive in one direction or another ( performance vs memory management ) retain the ability to revert to prior behavior by setting the `batch_rows` parameter to be equal to the number of rows of the `data.frame`.
* RE: snowflake specifically, going by the paragraph above, like their driver internally tries to optimize and select between streaming data vs file transfer and it seems to do this based on ```the number of values```.  From the way it is stated it sounds like using the files is slower than streaming the data, so maybe we stumbled across an optimization without necessarily targetting it ( again have to weigh this of making repeated calls versus one call with a large data-set ).
  * This is a bit out of our control and looking at the documentation for their driver I did not see an easy way to configure this behavior.  Though I did see a number of interesting `PUT` config options - for example `put_compresslv`.
* It is curious that the behavior  on Windows is different than Linux. Nothing on our end, from what i can tell, should be different.
